devito/finite_differences/finite_difference.py

```diff
@@ -198,8 +229,10 @@ def first_derivative(expr, **kwargs):
         ind = [(dim + i * diff) for i in range(-int(order / 2),
                                                int((order + 1) / 2) + 1)]
     # Finite difference weights from Taylor approximation with this positions
-    c = finite_diff_weights(1, ind, dim)
-    c = c[-1][-1]
+    if fd_weight_type is 'symbolic':
+        c = symbolic_weights(f, 1, indices, dim)
```

Comment:
and now that I think about it: I have a mild preference towards having ``finite_diff_weights`` accepting a further parameter "symbolic", rather than having two different functions ``symbolic_weights`` and ``fintie_diff_weights``. Because that way we could avoid a proliferation of if-then-else in all functions computing derivatives

Reply:
Sounds good. Since `finite_diff_weights` (as it is currently) is a sympy function we'll need our own wrapper with this addition.